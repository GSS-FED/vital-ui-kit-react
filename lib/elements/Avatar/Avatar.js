// @flow

import React, { type Node } from 'react';
import styled from 'styled-components';

import { AvatarBadge } from './';
import { avatarSizes } from './constants';
import DEFAULTAVATAR from '../../assets/img/avatar/default-avatar.svg';

const Root = styled.div`
  position: relative;
  display: inline-block;
`;

const Image = styled.img`
  width: ${props => avatarSizes[props.size].size};
  height: ${props => avatarSizes[props.size].size};
  border-radius: ${props =>
    props.round ? '50%' : avatarSizes[props.size].borderRadius};
  background-color: #f0f0f0;
  box-sizing: border-box;
`;

type SizeType = 'xlarge' | 'large' | 'medium' | 'small' | 'xsmall';

type Props = {
  src?: string,
  /** Default is false, check if you want ciruclar avatar */
  round?: boolean,
  size?: SizeType,
  badge?: string,
};

const Avatar = ({ src, round = false, size = 'medium', badge }: Props) => {
  const renderBadge = (): ?Node => {
    if (!badge) {
      return null;
    }

    return <AvatarBadge label={badge} size={size} round={round} />;
  };

  return (
    <Root>
      <Image src={src || DEFAULTAVATAR} size={size} round={round} />
      {renderBadge()}
    </Root>
  );
};

export default Avatar;
