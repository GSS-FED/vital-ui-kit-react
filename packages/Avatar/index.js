export { default } from './Avatar';
export { default as AvatarBadge } from './AvatarBadge';
