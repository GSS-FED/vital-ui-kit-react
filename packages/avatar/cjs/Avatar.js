'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _templateObject = _taggedTemplateLiteral(['\n  position: relative;\n  display: inline-block;\n'], ['\n  position: relative;\n  display: inline-block;\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  width: ', ';\n  height: ', ';\n  border-radius: ', ';\n  background-color: ', ';\n  box-sizing: border-box;\n'], ['\n  width: ', ';\n  height: ', ';\n  border-radius: ', ';\n  background-color: ', ';\n  box-sizing: border-box;\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _reactTheme = require('@vital-ui/react-theme');

var _AvatarBadge = require('./AvatarBadge');

var _AvatarBadge2 = _interopRequireDefault(_AvatarBadge);

var _constants = require('./constants');

var _defaultAvatar = require('../assets/default-avatar.svg');

var _defaultAvatar2 = _interopRequireDefault(_defaultAvatar);

var _defaultAvatarO = require('../assets/default-avatar-o.svg');

var _defaultAvatarO2 = _interopRequireDefault(_defaultAvatarO);

var _defaultAvatarM = require('../assets/default-avatar-m.svg');

var _defaultAvatarM2 = _interopRequireDefault(_defaultAvatarM);

var _defaultAvatarF = require('../assets/default-avatar-f.svg');

var _defaultAvatarF2 = _interopRequireDefault(_defaultAvatarF);

var _defaultAvatarMO = require('../assets/default-avatar-m-o.svg');

var _defaultAvatarMO2 = _interopRequireDefault(_defaultAvatarMO);

var _defaultAvatarFO = require('../assets/default-avatar-f-o.svg');

var _defaultAvatarFO2 = _interopRequireDefault(_defaultAvatarFO);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); } /**
                                                                                                                                                   * 
                                                                                                                                                   * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
                                                                                                                                                   * MIT license
                                                                                                                                                   */

var Root = _styledComponents2.default.div(_templateObject);

var Image = _styledComponents2.default.img(_templateObject2, function (_ref) {
  var size = _ref.size;
  return _constants.avatarSizes[size].size;
}, function (_ref2) {
  var size = _ref2.size;
  return _constants.avatarSizes[size].size;
}, function (_ref3) {
  var size = _ref3.size,
      round = _ref3.round;
  return round ? '50%' : _constants.avatarSizes[size].borderRadius;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.grey200;
});

Image.defaultProps = {
  theme: _reactTheme.defaultTheme
};

/**
 * @render react
 * @name Avatar
 * @description Vital UI Kit Avatar Component
 * @example
 * <Avatar
 *  badge="99+"
 *  size="large"
 *  gender="male"
 *  round
 * />
 */

var Avatar = function Avatar(_ref5) {
  var src = _ref5.src,
      round = _ref5.round,
      size = _ref5.size,
      badge = _ref5.badge,
      outline = _ref5.outline,
      gender = _ref5.gender,
      containerStyle = _ref5.containerStyle,
      imageStyle = _ref5.imageStyle,
      badgeStyle = _ref5.badgeStyle,
      props = _objectWithoutProperties(_ref5, ['src', 'round', 'size', 'badge', 'outline', 'gender', 'containerStyle', 'imageStyle', 'badgeStyle']);

  var renderBadge = function renderBadge() {
    if (!badge) {
      return null;
    }

    return _react2.default.createElement(_AvatarBadge2.default, {
      label: badge,
      size: size,
      round: round,
      style: badgeStyle
    });
  };

  var renderDefaultAvatar = function renderDefaultAvatar() {
    if (src) {
      return src;
    }
    if (outline && gender) {
      if (gender === 'female') return _defaultAvatarFO2.default;
      if (gender === 'male') return _defaultAvatarMO2.default;
      return _defaultAvatarO2.default;
    }
    if (!outline && gender) {
      if (gender === 'female') return _defaultAvatarF2.default;
      if (gender === 'male') return _defaultAvatarM2.default;
      return _defaultAvatar2.default;
    }
    if (outline && !gender) {
      return _defaultAvatarO2.default;
    }
    return _defaultAvatar2.default;
  };

  return _react2.default.createElement(
    Root,
    _extends({ style: containerStyle }, props),
    _react2.default.createElement(Image, _extends({
      src: renderDefaultAvatar(),
      size: size,
      round: round,
      style: imageStyle
    }, props)),
    renderBadge()
  );
};

Avatar.defaultProps = {
  badge: null,
  src: null,
  gender: undefined,
  round: false,
  size: 'medium',
  outline: false,
  imageStyle: null,
  containerStyle: null,
  badgeStyle: null
};

exports.default = Avatar;