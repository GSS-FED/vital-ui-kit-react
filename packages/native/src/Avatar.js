// @flow
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable global-require */

import React from 'react';
import { ImageStyle, ViewStyle } from 'react-native';
import styled from 'styled-components/native';
import boxShadow from './helper/boxShadow';

const AVATAR_SIZE = 37;

const Root = styled.View`
  height: ${props => props.size};
  width: ${props => props.size};
  ${props =>
    props.shadow &&
    `
    ${boxShadow(-14, 8, 35, '#000', 0.34)}
  `};
`;

const Image = styled.Image`
  width: 100%;
  height: 100%;
  border-radius: ${props => props.size / 2};
`;

export type AvatarProps = {
  size?: number,
  src?: string,
  shadow: boolean,
  wrapperStyle?: ViewStyle,
  imageStyle?: ImageStyle,
};

const Avatar = ({
  size = AVATAR_SIZE,
  src = null,
  shadow = false,
  wrapperStyle,
  imageStyle,
  ...props
}: AvatarProps) => (
  <Root shadow={shadow} size={size} style={wrapperStyle} {...props}>
    <Image
      source={
        src ? { uri: src } : require('../assets/avatar-alien.png')
      }
      size={size}
      style={imageStyle}
    />
  </Root>
);

export default Avatar;
