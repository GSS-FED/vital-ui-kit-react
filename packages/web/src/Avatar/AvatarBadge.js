// @flow

import React from 'react';
import styled, { css } from 'styled-components';

import { trunTo } from '../utils/math';
import { badgeSizes } from './constants';

const badgeStyle = props => {
  if (props.size === 'xlarge') {
    return css`
      padding: 0 5px;
      border-radius: 0.75rem;
      font-size: 0.725rem;
      left: ${props.round ? '70%' : 'auto'};
      right: ${props.round ? 'auto' : '-12%'};
      top: ${props.round ? 0 : `calc(${badgeSizes[props.size].height} / -2)`};
    `;
  }
  // circle badge style
  return css`
    padding: 0;
    border-radius: 50%;
    font-size: 0;
    width: ${badgeSizes[props.size].height};
    height: ${badgeSizes[props.size].height};
    right: calc(${badgeSizes[props.size].height} / -2);
    top: ${props.round ? 0 : `calc(${badgeSizes[props.size].height} / -2)`};
    left: ${props.round ? '70%' : 'auto'};
    right: ${props.round ? 'auto' : badgeSizes[props.size].height / -2};

    &:before {
      content: '';
      position: absolute;
      top: calc(50% - (3px / 2));
      left: calc(50% - (3px / 2));
      background: ${props.label ? '#fff' : 'transparent'};
      border-radius: 50%;
      width: 3px;
      height: 3px;
    }
  `;
};

const Root = styled.span`
  position: absolute;
  background-color: #eb5000;
  color: #fff;
  vertical-align: middle;
  height: ${props => badgeSizes[props.size].height};
  line-height: ${props => badgeSizes[props.size].height};
  ${props => badgeStyle(props)};
`;

type Props = {
  label: string,
  size?: string,
  round?: boolean
};

const AvatarBadge = ({ label, size, round }: Props) => (
  <Root label={label} size={size} round={round}>
    {size === 'xlarge' ? trunTo(label) : null}
  </Root>
);

export default AvatarBadge;
