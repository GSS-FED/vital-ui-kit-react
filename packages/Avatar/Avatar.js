// @flow

import React, { type Node } from 'react';
import styled from 'styled-components';

import AvatarBadge from './AvatarBadge';
import { avatarSizes } from './constants';
import DEFAULT_AVATAR from './assets/default-avatar.svg';
import DEFAULT_AVATAR_OUTLINE from './assets/default-avatar-o.svg';
import DEFAULT_AVATAR_M from './assets/default-avatar-m.svg';
import DEFAULT_AVATAR_F from './assets/default-avatar-f.svg';
import DEFAULT_AVATAR_M_O from './assets/default-avatar-m-o.svg';
import DEFAULT_AVATAR_F_O from './assets/default-avatar-f-o.svg';

const Root = styled.div`
  position: relative;
  display: inline-block;
`;

const Image = styled.img`
  width: ${props => avatarSizes[props.size].size};
  height: ${props => avatarSizes[props.size].size};
  border-radius: ${props =>
    props.round ? '50%' : avatarSizes[props.size].borderRadius};
  background-color: #f0f0f0;
  box-sizing: border-box;
`;

type SizeType = 'xlarge' | 'large' | 'medium' | 'small' | 'xsmall';
type Gender = 'male' | 'female';

type Props = {
  src?: string,
  round?: boolean,
  size?: SizeType,
  badge?: string,
  gender?: Gender,
  outline?: boolean
};

/**
 * @render react
 * @name Avatar
 * @description Avatar UI Component
 * @example
 * <Avatar
 *  badge="99+"
 *  size="large"
 *  gender="male"
 *  round
 * />
 */

const Avatar = ({
  src,
  round,
  size,
  badge,
  outline,
  gender,
  ...props
}: Props) => {
  const renderBadge = (): ?Node => {
    if (!badge) {
      return null;
    }

    return <AvatarBadge label={badge} size={size} round={round} />;
  };

  const renderDefaultAvatar = (): string => {
    if (src) {
      return src;
    }
    if (outline && gender) {
      if (gender === 'female') return DEFAULT_AVATAR_F_O;
      if (gender === 'male') return DEFAULT_AVATAR_M_O;
      return DEFAULT_AVATAR_OUTLINE;
    } else if (!outline && gender) {
      if (gender === 'female') return DEFAULT_AVATAR_F;
      if (gender === 'male') return DEFAULT_AVATAR_M;
      return DEFAULT_AVATAR;
    }
    if (outline && !gender) {
      return DEFAULT_AVATAR_OUTLINE;
    }
    return DEFAULT_AVATAR;
  };

  return (
    <Root>
      <Image src={renderDefaultAvatar()} size={size} round={round} {...props} />
      {renderBadge()}
    </Root>
  );
};

Avatar.defaultProps = {
  round: false,
  size: 'medium',
  outline: false
};

export default Avatar;
