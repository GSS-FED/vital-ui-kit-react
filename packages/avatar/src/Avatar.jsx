/**
 * @flow
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import React, { type Node } from 'react';
import styled from 'styled-components';

import AvatarBadge from './AvatarBadge';
import { avatarSizes } from './constants';
import DEFAULT_AVATAR from '../assets/default-avatar.svg';
import DEFAULT_AVATAR_OUTLINE from '../assets/default-avatar-o.svg';
import DEFAULT_AVATAR_M from '../assets/default-avatar-m.svg';
import DEFAULT_AVATAR_F from '../assets/default-avatar-f.svg';
import DEFAULT_AVATAR_M_O from '../assets/default-avatar-m-o.svg';
import DEFAULT_AVATAR_F_O from '../assets/default-avatar-f-o.svg';

const Root = styled.div`
  position: relative;
  display: inline-block;
`;

const Image = styled.img`
  width: ${({ size }) => avatarSizes[size].size};
  height: ${({ size }) => avatarSizes[size].size};
  border-radius: ${({ size, round }) =>
    round ? '50%' : avatarSizes[size].borderRadius};
  background-color: ${({ theme }) => theme.grey200};
  box-sizing: border-box;
`;

type SizeType = 'xlarge' | 'large' | 'medium' | 'small' | 'xsmall';
type Gender = 'male' | 'female';

type Props = {
  src?: string,
  round?: boolean,
  size?: SizeType,
  badge?: string,
  gender?: Gender,
  outline?: boolean,
  containerStyle?: CSSStyleDeclaration,
  imageStyle?: CSSStyleDeclaration,
  badgeStyle?: CSSStyleDeclaration,
};

/**
 * @render react
 * @name Avatar
 * @description Vital UI Kit Avatar Component
 * @example
 * <Avatar
 *  badge="99+"
 *  size="large"
 *  gender="male"
 *  round
 * />
 */

const Avatar = ({
  src,
  round,
  size,
  badge,
  outline,
  gender,
  containerStyle,
  imageStyle,
  badgeStyle,
  ...props
}: Props) => {
  const renderBadge = (): ?Node => {
    if (!badge) {
      return null;
    }

    return (
      <AvatarBadge
        label={badge}
        size={size}
        round={round}
        style={badgeStyle}
      />
    );
  };

  const renderDefaultAvatar = (): string => {
    if (src) {
      return src;
    }
    if (outline && gender) {
      if (gender === 'female') return DEFAULT_AVATAR_F_O;
      if (gender === 'male') return DEFAULT_AVATAR_M_O;
      return DEFAULT_AVATAR_OUTLINE;
    }
    if (!outline && gender) {
      if (gender === 'female') return DEFAULT_AVATAR_F;
      if (gender === 'male') return DEFAULT_AVATAR_M;
      return DEFAULT_AVATAR;
    }
    if (outline && !gender) {
      return DEFAULT_AVATAR_OUTLINE;
    }
    return DEFAULT_AVATAR;
  };

  return (
    <Root style={containerStyle} {...props}>
      <Image
        src={renderDefaultAvatar()}
        size={size}
        round={round}
        style={imageStyle}
        {...props}
      />
      {renderBadge()}
    </Root>
  );
};

Avatar.defaultProps = {
  badge: null,
  src: null,
  gender: undefined,
  round: false,
  size: 'medium',
  outline: false,
  imageStyle: null,
  containerStyle: null,
  badgeStyle: null,
};

export default Avatar;
