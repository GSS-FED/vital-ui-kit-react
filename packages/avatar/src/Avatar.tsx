/**
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import styled, { css } from 'styled-components';
import { defaultTheme } from '@vital-ui/react-theme';
import cn from 'classnames';

import { AvatarBadge } from './AvatarBadge';
import { avatarSizes, Size } from './constants';
import { defaultAvatarSets } from './default-avatar';

type SizeStyleProps = {
  size: Size;
  circle?: boolean;
  builtinTheme?: typeof builtinTheme;
};

const sizeStyle = css<SizeStyleProps>`
  width: ${({ size }) => builtinTheme[size].size};
  height: ${({ size }) => builtinTheme[size].size};
  border-radius: ${({ size, circle }) =>
    circle ? '50%' : builtinTheme[size].borderRadius};
`;

const Root = styled.div`
  position: relative;
  display: inline-block;
`;

const Image = styled.img<SizeStyleProps>`
  background-color: ${({ theme }) => theme.grey200};
  box-sizing: border-box;
  ${sizeStyle};
`;

const ImageWrapper = styled.div`
  overflow: hidden;
  ${sizeStyle};
`;

Image.defaultProps = {
  theme: defaultTheme,
};

const builtinTheme = {
  ...avatarSizes,
};

type Gender = 'male' | 'female';

export interface AvatarProps {
  /** Default sets of avatar */
  bulltinAvatars?: typeof defaultAvatarSets;
  /** Each Avatar size and borderRadius if circle  */
  builtinTheme?: typeof builtinTheme;
  /** Image src html attr of the avatar. */
  src?: string;
  /** @deprecated Circle style. */
  round?: boolean;
  /** Circle style. */
  circle?: boolean;
  /** Avatar size, default is `medium`, `xlarge`, `large`, `medium`, `small`, `xsmall` */
  size?: Size;
  /** Value for the right top badge */
  badge?: React.ReactNode;
  gender?: Gender;
  outline?: boolean;
  containerStyle?: React.CSSProperties;
  imageStyle?: React.CSSProperties;
  badgeStyle?: React.CSSProperties;
  /** Default is `vital__avatar` */
  containerClassName?: string;
  /** Default is `vital__avatar-image` */
  imageClassName?: string;
  /** Default is `vital__avatar-badge` */
  badgeClassName?: string;
}

/**
 * @render react
 * @name Avatar
 * @description Vital UI Kit Avatar Component
 * @example
 * <Avatar
 *  badge="99+"
 *  size="large"
 *  gender="male"
 *  round
 * />
 */
export class Avatar extends React.Component<AvatarProps> {
  static defaultProps = {
    bulltinAvatars: defaultAvatarSets,
    builtinTheme,
    badge: null,
    src: null,
    gender: undefined,
    circle: false,
    size: 'medium',
    outline: false,
    imageStyle: undefined,
    containerStyle: undefined,
    badgeStyle: undefined,
    containerClassName: '',
    imageClassName: '',
    badgeClassName: '',
  };

  static Badge: typeof AvatarBadge = AvatarBadge;

  render() {
    const {
      src,
      round,
      circle,
      size = 'medium',
      badge,
      outline,
      gender,
      containerStyle,
      imageStyle,
      badgeStyle,
      containerClassName,
      imageClassName,
      badgeClassName,
      ...props
    } = this.props;
    return (
      <Root
        style={containerStyle}
        className={cn('vital__avatar', containerClassName)}
        {...props}
      >
        {src ? (
          <Image
            className={cn('vital__avatar-image', imageClassName)}
            src={src}
            size={size!}
            circle={circle}
            style={imageStyle}
            {...props}
          />
        ) : (
          <ImageWrapper
            className={cn('vital__avatar-image-wrapper')}
            size={size}
            circle={circle}
          >
            {this.renderDefaultAvatar()}
          </ImageWrapper>
        )}
        {this.renderBadge()}
      </Root>
    );
  }

  private renderDefaultAvatar = (): JSX.Element => {
    const {
      outline,
      gender,
      bulltinAvatars,
      size,
      builtinTheme,
    } = this.props;
    const defaultAvatarRenderer = bulltinAvatars || defaultAvatarSets;
    const avatarProps = {
      width: builtinTheme![size!].size,
      height: builtinTheme![size!].size,
    };
    if (outline && gender) {
      if (gender === 'female') {
        return defaultAvatarRenderer.femaleOutline(avatarProps);
      }
      if (gender === 'male') {
        return defaultAvatarRenderer.maleOutLine(avatarProps);
      }
      return defaultAvatarRenderer.outline(avatarProps);
    }
    if (!outline && gender) {
      if (gender === 'female') {
        return defaultAvatarRenderer.female(avatarProps);
      }
      if (gender === 'male') {
        return defaultAvatarRenderer.male(avatarProps);
      }
      return defaultAvatarRenderer.default(avatarProps);
    }
    if (outline && !gender) {
      return defaultAvatarRenderer.outline(avatarProps);
    }
    return defaultAvatarRenderer.default(avatarProps);
  };

  private renderBadge = () => {
    const {
      badge,
      size,
      round,
      circle = round,
      badgeStyle,
      badgeClassName,
    } = this.props;
    if (!badge) {
      return null;
    }
    if (typeof badge === 'string' || typeof badge === 'number') {
      return (
        <AvatarBadge
          className={cn('vital__avatar-badge', badgeClassName)}
          label={badge}
          size={size}
          circle={circle}
          style={badgeStyle}
        />
      );
    }
    return badge;
  };
}
