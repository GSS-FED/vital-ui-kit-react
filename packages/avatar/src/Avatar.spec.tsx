import React from 'react';
import renderer from 'react-test-renderer';
import { Avatar } from './Avatar';
import { ThemeProvider } from '@vital-ui/react-theme';

describe('Avatar', () => {
  it('should render with default styles', () => {
    const avatar = renderer.create(<Avatar />);
    expect(avatar.toJSON()).toMatchSnapshot();
  });

  it('should render with number size prop', () => {
    const avatar = renderer.create(<Avatar size={150} />);
    expect(avatar.toJSON()).toMatchSnapshot();
  });

  it('should render with css prop', () => {
    const avatar = renderer.create(
      <Avatar
        size={150}
        css={`
          border: 1px solid grey;
        `}
        circle
      />,
    );

    expect(avatar.toJSON()).toMatchSnapshot();
  });

  it('should render with theme props', () => {
    const avatar = renderer.create(
      <ThemeProvider
        componentTheme={props => ({
          avatar: {
            xsmall: {
              size: '20px',
              borderRadius: '3px',
            },
          },
        })}
      >
        <Avatar size="xsmall" />
      </ThemeProvider>,
    );
    expect(avatar.toJSON()).toMatchSnapshot();
  });
});
