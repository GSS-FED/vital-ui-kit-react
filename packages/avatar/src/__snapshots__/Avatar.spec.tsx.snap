// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render with default styles 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

.c1 {
  overflow: hidden;
  width: 32px;
  height: 32px;
  border-radius: 3px;
}

<div
  className="c0 vital__avatar"
>
  <div
    className="c1 vital__avatar-image-wrapper"
    size="medium"
  >
    <svg
      height="32px"
      viewBox="0 0 124 124"
      width="32px"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h124v124H0z"
          fill="#7189B6"
        />
        <circle
          cx={62}
          cy={41}
          fill="#FFF"
          r={18}
        />
        <path
          d="M28 102c0-19.882 15.222-36 34-36s34 16.118 34 36H28z"
          fill="#FFF"
        />
      </g>
    </svg>
  </div>
</div>
`;

exports[`Avatar should render with number size prop 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

.c1 {
  overflow: hidden;
  width: 150px;
  height: 150px;
  border-radius: 3px;
}

<div
  className="c0 vital__avatar"
>
  <div
    className="c1 vital__avatar-image-wrapper"
    size={150}
  >
    <svg
      height={150}
      viewBox="0 0 124 124"
      width={150}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h124v124H0z"
          fill="#7189B6"
        />
        <circle
          cx={62}
          cy={41}
          fill="#FFF"
          r={18}
        />
        <path
          d="M28 102c0-19.882 15.222-36 34-36s34 16.118 34 36H28z"
          fill="#FFF"
        />
      </g>
    </svg>
  </div>
</div>
`;

exports[`Avatar should render with theme props 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

.c1 {
  overflow: hidden;
  width: 20px;
  height: 20px;
  border-radius: 3px;
}

<div
  className="c0 vital__avatar"
>
  <div
    className="c1 vital__avatar-image-wrapper"
    size="xsmall"
  >
    <svg
      height="16px"
      viewBox="0 0 124 124"
      width="16px"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h124v124H0z"
          fill="#7189B6"
        />
        <circle
          cx={62}
          cy={41}
          fill="#FFF"
          r={18}
        />
        <path
          d="M28 102c0-19.882 15.222-36 34-36s34 16.118 34 36H28z"
          fill="#FFF"
        />
      </g>
    </svg>
  </div>
</div>
`;
